index.js

```diff
@@ -317,43 +387,48 @@ function updateBadge() {
   }
 }
 
-// reset the current hang stacks so we only show the new ones coming in
-mostRecentHangs();
-recentHangs = [];
-
-const CHECK_FOR_HANG_INTERVAL = 400; // in millis
-let { numHangs: numHangs, minBucketLowerBound: computedThreshold } = numGeckoHangs(); // note: this will be null if the hang counter is not available
-let baseNumHangs = 0; // the number of hangs at the time the counter was last reset
-setInterval(() => {
-  let { numHangs: hangCount, minBucketLowerBound: lower } = numGeckoHangs();
-  if (hangCount !== numHangs) {
-    numHangs = hangCount;
-    updateBadge();
-    let hangs = mostRecentHangs();
-    panel.port.emit("set-hangs", hangs);
-    if (hangs.length > 0) {
-      button.label = "Most recent hang stack:\n\n" + hangs[hangs.length - 1].stack;
-    } else {
-      button.label = "No recent hang stacks.";
-    }
-    //exports.observe(undefined, "thread-hang");
-  }
-  if (lower !== computedThreshold) { // update the computed threshold
-    computedThreshold = lower;
-    panel.port.emit("set-computed-threshold", computedThreshold);
-  }
-}, CHECK_FOR_HANG_INTERVAL);
-clearCount();
-
 function clearCount() {
   baseNumHangs = numHangs;
   numHangsObserved = 0;
+  computedThreshold = 0;
+  cachedRecentHangs = []; // empty out the list of hangs
   updateBadge();
   panel.port.emit("set-computed-threshold", computedThreshold);
   panel.port.emit("set-hangs", []); // clear the panel's list of hangs
-  recentHangs = []; // empty out the list of hangs
 }
 
+const CHECK_FOR_HANG_INTERVAL = 400; // in millis
+function update() {
+  getHangs().then((result) => {
+    let { numHangs: hangCount, minBucketLowerBound: lower } = result; // note: numHangs will be null if the hang counter is not available
```

Comment:
You can decompose object arguments in the parameter list, I think. (Yes you can, see [this MDN](https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment#ES6_version))

So... something like 
```js
getHangs().then(({numHangs: hangCount, minBucketLowerBound: lower}) => {
...
```